app/frontend/src/components/admin/inspection/LocationEdit.vue

```diff
@@ -0,0 +1,199 @@
+          throw new Error('No response data from API while submitting update');
+        }
+      } catch (error) {
+        console.error(`Error submitting update: ${error}`); // eslint-disable-line no-console
+      }
+    }
+
+  }
+})
+
+export default {
+  name: 'LocationEdit',
+  components: {
+    Location,
+  },
+  props: {
+    ipcPlanId: {
+      required: true,
+      type: String
+    },
+  },
+  data() {
+    return {
+      showEditBtn: false,
+      dialog: false,
+      updateValid: false,
+      reviewMode: false,
+      startDateMenu: false,
+      endDateMenu: false,
+    };
+  },
+  computed: {
+    ...mapGetters('form', ['business', 'ipcPlan', 'location', 'submissionComplete']),
+    ...mapGetters('auth', ['hasSilvipcRoles']),
+
+    // Business data
+    businessName: {
+      get() { return this.business.name; },
+    },
+
+    // location data
+    startDate: {
+      get() { return this.location.startDate; },
+    },
+    endDate: {
+      get() { return this.location.endDate; },
+    },
+    locationCity: {
+      get() { return this.location.city; },
+    },
+    cityLatitude: {
+      get() { return this.location.cityLatitude; },
+    },
+    cityLongitude: {
+      get() { return this.location.cityLongitude; },
+    },
+    numberOfWorkers: {
+      get() { return this.location.numberOfWorkers ? this.location.numberOfWorkers.toString() : ''; },
+    },
+    accTents: {
+      get() { return this.location.accTents; },
+    },
+    tentDetails: {
+      get() { return this.location.tentDetails; },
+    },
+    accMotel: {
+      get() { return this.location.accMotel; },
+    },
+    motelName: {
+      get() { return this.location.motelName; },
+    },
+    motelAddressLine1: {
+      get() { return this.location.motelAddressLine1; },
+    },
+    motelAddressLine2: {
+      get() { return this.location.motelAddressLine2; },
+    },
+    motelCity: {
+      get() { return this.location.motelCity; },
+    },
+    motelProvince: {
+      get() { return this.location.motelProvince; },
+    },
+    motelPostalCode: {
+      get() { return this.location.motelPostalCode; },
+    },
+    accWorkersHome: {
+      get() { return this.location.accWorkersHome; },
+    },
+  },
+  methods: {
+
+    async update() {
+      if(this.$refs.form.validate()) {
+        await this.updateForm();
+        if (this.submissionComplete) {
+          this.dialog = false;
+        }
+      } else {
+        const modal = document.querySelector('.v-dialog--active');
+        modal.scrollBy(0, -500);
+      }
+    },
+
+    async cancelUpdate(){
+      // TODO reload location details in step2 component in case user changed details but didnt submit update
+      this.dialog = false;
+    },
+
+  },
+  mounted() {
+    // show edit button for inspectors on submission review page
+    this.showEditBtn = (this.$route.name == 'Submission' && this.hasSilvipcRoles([SilvipcRoles.INSPECTOR])) ? true : false;
+  },
+  created() {
```

Comment:
not sure why i need to lazy load a store if it's already defined in this component.

Reply:
You want to lazy-load and unload a store because this allows you to control _when_ a namespaced module store is effective and mounted or not based on the visibility and lifecycle of a component (or set of them). We want to bind the lifecycle of your 'localStore" module to just this component, but also be able to continue to use the business logic in that module.